oauth2_provider/oauth2_validators.py

```diff
@@ -877,22 +796,54 @@ def validate_id_token(self, token, scopes, request):
         if not token:
             return False
 
-        key = jwk.JWK.from_pem(oauth2_settings.OIDC_RSA_PRIVATE_KEY.encode("utf8"))
+        id_token = self._load_id_token(token)
+        if not id_token:
+            return False
+
+        if not id_token.allow_scopes(scopes):
+            return False
+
+        request.client = id_token.application
+        request.user = id_token.user
+        request.scopes = scopes
+        # this is needed by django rest framework
+        request.access_token = id_token
+        return True
 
+    def _load_id_token(self, token):
+        key = self._get_key_for_token(token)
+        if not key:
+            return None
         try:
             jwt_token = jwt.JWT(key=key, jwt=token)
-            id_token = IDToken.objects.get(token=jwt_token.serialize())
```

Comment:
**issue**: I think we need to keep this since we are using token still. We can change it after the double write is live.

Reply:
Makes sense or else the current active tokens (if there is any) during the deploy window will not be valid anymore.